### ISSUE TITLE
Missing file: libpulse.so.0

### ISSUE THREAD

**Issue by mpelley** [docker-android-sdk/android-31#1]
I am trying to run the emulator so it displays the device.  (Some of my Espresso tests fail without the device display.)  So I am **not** supplying the -no-window parameter when starting the emulator.  When I don't specify -no-window, the emulator complains that the file libpulse.so.0 is missing and won't start.  If it matters, I am specifying `"system-images;android-26;google_apis_playstore;x86"` and device `-d 16` when I create the AVD.  (I eventually will be testing my app with a list of various API versions and various device types.)

**Comment by ppoffice** [comment]
@mpelley 
https://github.com/docker-android-sdk/android-30/issues/11#issuecomment-970354423

**Comment by ppoffice** [comment]
As a workaround, you may use `scrcpy` for screen output. You can also create a custom docker image on top of this one to install `libpulse`.

**Comment by rades101** [comment]
Hello,

I was trying to start an emulator without a -no-window option, to be able to see the display. Also, I installed  `libxcursor-dev` and `libpulse-dev`. Unfortunately, I am not able to start the emulator, when  `emulator -avd first_avd  -no-audio  &` executed I am getting 
```
WARNING | unexpected system image feature string, emulator might not function correctly, please try updating the emulator.
INFO    | Warning: could not connect to display  ((null):0, (null))

INFO    | Info: Could not load the Qt platform plugin "xcb" in "/opt/android-sdk-linux/emulator/lib64/qt/plugins" even though it was found. ((null):0, (null))

Fatal: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.
 ((null):0, (null))
INFO    | Fatal: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.
 ((null):0, (null))
```
Regarding `scrcpy` if emulator started with `emulator -avd first_avd -no-audio -no-window` command UI cannot be casted, since the emulator started in headless mode. So I am getting this error
`CRITICAL: Could not initialize SDL: No available video device`

For me, interaction with UI is needed just the first time during setting up the device.

Best,
Rade

**Comment by ppoffice** [comment]
@rades101 You should start the docker container with `--network host` and run `scrcpy` on your host OS instead of the container. Also, `--network host` does not work properly for docker on Windows and macOS. If your docker is running on such OSes, you may need to install a remote desktop server (such as VNC) in your container and connects to it.

**Comment by rades101** [comment]
@ppoffice thanks for the response!  I started the container with `--network host`, but if the emulator started with `emulator -avd first_avd -no-audio -no-window`, option `-no-window` is signal to start the emulator in headless mode, so `scrcpy` cannot connect to the emulator. And this is the problem that I described in my previous post. Looks there are issues in a container, and the emulator cannot be started in GUI mode.

**Comment by ppoffice** [comment]
@rades101 Where did you run `scrcpy`?

PS: I have no problem using `scrcpy` with the headless emulator:

<img width="1442" alt="Screenshot 2022-03-15 134707" src="https://user-images.githubusercontent.com/8849362/158439911-84f69a27-c8a7-48b6-83e5-2579dc321737.png">

**Comment by rades101** [comment]
@ppoffice I run `scrcpy` from the host. But I got a different error message after I followed your steps from a screenshot. 
<img width="1545" alt="Screen Shot 2022-03-16 at 12 23 33 AM" src="https://user-images.githubusercontent.com/29555268/158516713-65639c3e-fedd-4d12-a2db-71ae980dd215.png">

Btw, for me takes forever to boot a device, what is your hardware specification for ubuntu? In my case, I have 4 CPUs and 8GB of RAM.
